store.go

```diff
@@ -1932,10 +1945,46 @@ func (s *store) Mount(id, mountLabel string) (string, error) {
 	if modified, err := rlstore.Modified(); modified || err != nil {
 		rlstore.Load()
 	}
-	if rlstore.Exists(id) {
-		return rlstore.Mount(id, mountLabel)
+	if !rlstore.Exists(id) {
+		return "", ErrLayerUnknown
 	}
-	return "", ErrLayerUnknown
+
+	rootUID, rootGID, err := idtools.GetRootUIDGID(s.uidMap, s.gidMap)
+	if err != nil {
+		return "", err
+	}
+
+	merged := ""
+	if rootUID != 0 {
```

Comment:
Why have two if rootUID blocks?  Can't this just be done later after the rlstore.Mount()

Reply:
I preferred to keep the Mount operation as late as possible, so if we fail earlier we don't need to Mount and Unmount for the cleanup